### PR TITLE
Sqroot and Error handlers added; comments incorporated.

### PR DESCRIPTION
**Pull request by anthonysaya**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,61 +5,86 @@ window.addEventListener("load", () => {
...
       } else {
         return;
       }
     });
   }
 
   numButtons[10].addEventListener("click", () => {
-    if (currentString.includes(".")) {
+    if (errLock == true) {
+      return;
```
Comment (JRonkin) [inline index.js:32]: If you press the decimal button after an error, it should start typing a new number starting with "0."

File: index.js [hunk 2]
```diff
@@ -5,61 +5,86 @@ window.addEventListener("load", () => {
...
 
   numButtons[11].addEventListener("click", () => {
-    if (currentString.includes("-")) {
-      currentNum = Number(currentString) * -1;
-      numDisplay.textContent = currentNum.toString();
-      currentString = numDisplay.textContent;
+    if (errLock == true || currentString == "" || currentString == ".") {
+      return;
+    } else if (currentString.length > 10) {
+      currentString = (Number(currentString) * -1).toExponential(5);
```
Comment (JRonkin) [inline index.js:45]: I don't think the negative symbol should count towards the digit limit, especially if you have to rewrite the number in exponential format with less precision. On the same topic, the decimal point probably shouldn't count towards the limit either. What if you check the length of the string after removing all non-numeric characters?

File: index.js [hunk 3]
```diff
@@ -69,39 +94,49 @@ window.addEventListener("load", () => {
       numDisplay.textContent = currentString;
     } else if (firstNum != "" && currentString != "") {
       if (currentOp == "add") {
-        numDisplay.textContent = (firstNum + currentNum)
+        currentString = (firstNum + Number(currentString))
           .toString()
           .substring(0, 10);
```
Comment (JRonkin) [inline index.js:99]: Taking the substring of added numbers could result in an incorrect answer, like 5000000000 + 5000000000 = 1000000000. Why not trim it with exponential notation like you do for multiplication?

File: index.js [hunk 4]
```diff
@@ -69,39 +94,49 @@ window.addEventListener("load", () => {
...
-        currentString = numDisplay.textContent;
-        currentNum = Number(currentString);
+        numDisplay.textContent = currentString;
         firstNum = "";
         currentOp = "";
       } else if (currentOp == "subtract") {
-        numDisplay.textContent = (firstNum - currentNum)
+        currentString = (firstNum - Number(currentString))
           .toString()
           .substring(0, 10);
```
Comment (JRonkin) [inline index.js:106]: Same here

File: index.js [hunk 5]
```diff
@@ -69,39 +94,49 @@ window.addEventListener("load", () => {
...
+        if (Number(currentString) == 0) {
+          currentString = "Error";
+          numDisplay.textContent = currentString;
+          firstNum = "";
+          currentOp = "";
+          errLock = true;
+        } else {
+          currentString = (firstNum / Number(currentString))
+            .toString()
+            .substring(0, 10);
```
Comment (JRonkin) [inline index.js:129]: Here again, you should handle the overflow in a better way than taking a substring every time. If you're dividing by a very small decimal, you'll get a large number as a result, and cutting off the last digits will make the answer significantly wrong.

File: index.js (hunk 2 above)
Comment (anthonysaya) [reply index.js:45]: Updated in the next PR (albeit with a really gross regEx string for now)

File: index.js (hunk 3 above)
Comment (anthonysaya) [reply index.js:99]: Fixed in the next PR (I think)

File: index.js (hunk 4 above)
Comment (anthonysaya) [reply index.js:106]: Fixed in the next PR (I think)

File: index.js (hunk 5 above)
Comment (anthonysaya) [reply index.js:129]: Fixed in the next PR (I think)

File: index.js (hunk 1 above)
Comment (anthonysaya) [reply index.js:32]: Fixed in next PR

**Pull request merged**